src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4241,6 +4241,52 @@ FCVT_SAFE(fcvt_l_d, d);
 
 #undef FCVT_SAFE
 
+void MacroAssembler::java_round_float(Register dst, FloatRegister src,
+                                      FloatRegister ftmp, Register tmp) {
+  Label done;
+  assert_different_registers(dst, tmp);
+  mv(dst, zr);
+
+  fclass_s(tmp, src);
+
+  // dst = 0
+  // if +/-0, +/-subnormal numbers, signaling/quiet NaN
+  andi(tmp, tmp, 0b1100111100);
```

Comment:
Please, update this line ( and same in doubles) for new scheme of working with fclass mask ( https://github.com/openjdk/jdk/pull/16362/files#diff-314214875276cd9a11ecdfd52b68403ded286710ba0820461b0b510506f61a33R1077 )

Reply:
Fixed. Thx!